tests/std/tests/GH_002030_asan_annotate_vector/test.cpp

```diff
@@ -274,6 +274,93 @@ struct implicit_allocator : custom_test_allocator<T, Pocma, Stateless> {
 STATIC_ASSERT(_Container_allocation_minimum_asan_alignment<vector<char, implicit_allocator<char>>> == 1);
 STATIC_ASSERT(_Container_allocation_minimum_asan_alignment<vector<wchar_t, implicit_allocator<wchar_t>>> == 2);
 
+// Helper class for `ArenaAllocator`, where data is linearly allocated in a finite buffer.
+class Arena {
+public:
+    Arena(size_t allocation_size, size_t capacity):
+        data(new char[capacity* allocation_size]),
+        allocation_size(allocation_size),
+        capacity(capacity),
+        offset(0) {}
+
+    Arena(const Arena& other):
+        data(other.data),
+        allocation_size(other.allocation_size),
+        capacity(other.capacity),
+        offset(other.offset) {}
+
+    void* allocate(size_t n) {
+        assert(offset + n <= capacity && "Allocation failed: the arena is out of memory. You may call `reset` to free up space.");
+        void* result = data + (offset * allocation_size);
+        offset += n;
+        return result;
+    }
+
+    // Deallocates memory in the arena, and `memset`s it all to zero
+    void reset() noexcept {
+        offset = 0;
+        memset(data, 0, capacity*allocation_size);
+    }
+
+    char* const data;
+    const size_t allocation_size;
+    const size_t capacity;
+    size_t offset;
+};
+
+// An allocator that allocates memory in a pre-allocated buffer (the arena).
+template <typename T>
+class ArenaAllocator {
+public:
+
+    using value_type = T;
+    Arena* arena;
+
+    ArenaAllocator(size_t alloc_size, size_t capacity) {
+        assert(sizeof(T) <= alloc_size && "Constructor failed: allocation size is too small for target type.");
+        arena = new Arena(alloc_size, capacity);
+    }
+
+    template <class U>
+    ArenaAllocator(const ArenaAllocator<U>& other) {
+        assert(sizeof(U) <= arena->allocation_size && "Constructor failed: allocation size is too small for target type.");

```

Comment:
Actually, this should be `sizeof(T)`. Seems I forgot to push that change. Will fix that shortly

Reply:
addressed, resolving